xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/DecoratedList.java

```diff
@@ -0,0 +1,84 @@
+package com.xpn.xwiki.doc;
+
+import java.util.ArrayList;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class DecoratedList extends ArrayList<XWikiAttachment>
```

Comment:
I would call that XWikiAttachmentList to be more explicit and put it in an *.internal.* package to be sure it's not public API.